streaming/base/storage/upload.py

```diff
@@ -168,11 +168,14 @@ def __init__(self,
 
         if os.path.exists(self.local) and len(os.listdir(self.local)) != 0:
             if not exist_ok:
-                raise FileExistsError(f'Directory is not empty: {self.local}')
+                raise FileExistsError(
+                    f'{self.local} not empty. Set exist_ok to True to overwrite the current contents (will delete).'
+                )
             else:
                 logger.warning(
-                    f'Directory {self.local} exists and not empty. But continue to mkdir since exist_ok is set to be True.'
+                    f'Directory {self.local} exists and is not empty; exist_ok is set to True so will remove contents.'
                 )
+                shutil.rmtree(self.local)
```

Comment:
This will break all the other code that does not perform writing.

Reply:
Thank you for the input, I'll change the implementation now.

Reply:
Okay, I've moved the removal code into writer which should fix the issue.